src/policy/policy.cpp

```diff
@@ -141,7 +141,7 @@ bool IsStandardTx(const CTransaction& tx, const std::optional<unsigned>& max_dat
         else if ((whichType == TxoutType::MULTISIG) && (!permit_bare_multisig)) {
             reason = "bare-multisig";
             return false;
-        } else if (whichType != TxoutType::ANCHOR && IsDust(txout, dust_relay_fee)) {
+        } else if ((!permit_ephemeral_anchor || whichType != TxoutType::ANCHOR) && IsDust(txout, dust_relay_fee)) {
```

Comment:
Shouldn't this be:

```c++
} else if (whichType != ANCHOR && IsDust) {
    reason = "dust"; return false;
} else if (!permit_ephemeral_anchor && whichType == ANCHOR) {
    reason = "ephemeral-anchor"; return false;
}
```

(otherwise, even if you set `-ephermalanchors=0` you permit low-feerate relay/mining of large txs if they set their fee to 0 and include a non-dust OP_TRUE output)

Reply:
done